src/Polysemy/Scoped.hs

```diff
@@ -82,6 +88,74 @@ interpretScopedAs ::
   InterpreterFor (Scoped param resource effect) r
 interpretScopedAs resource =
   interpretScoped \ p use -> use =<< resource p
+{-# inline interpretScopedAs #-}
+
+-- |Higher-order interpreter for 'Scoped' that allows the handler to use additional effects that are interpreted by the
+-- resource allocator.
+--
+-- /Note/: It is necessary to specify the list of local interpreters with a type application; GHC won't be able to
+-- figure them out from the type of @withResource@:
+--
+-- > interpretScopedWithH @[AtomicState Int, Reader Bool] withResource \ _ -> \case
+-- >   SomeAction -> atomicPut . (> 0) =<< ask
```

Comment:
I'm not very clear on where these additional effects are interpreted in this example. Can you point it out?

Reply:
I planned to write a better example later, but I added `withResource` for you now. the types were also wrong :sweat_smile:

Reply:
Make a higher-order example so tactical actually gets some use! And have a first-order example for `interpretScopedWith`.

Reply:
I adapted [this one](https://git.tryp.io/tek/ribosome/src/branch/main/packages/host/lib/Ribosome/Host/Interpreter/MState.hs#L56) for an example.
It would probably look nicer if I could also use `Lock` there, but it's not a core effect.

Don't know whether it's such a good example because the local effect part is purely for expressivity purposes – I could just as well use a raw `TVar` here.

Reply:
to really drive the point home you could make both `Lock` and `AtomicState` scoped effects as well, then there wouldn't be any `Embed IO` in the interpreter :sweat_smile: would also showcase the possibility of using `Scoped` in a resource allocator, which is pretty neat in my opinion